### PR TITLE
Qc report

### PR DESCRIPTION
**Pull request by DiDigsDNA**
Wanted to read in the analysis stage from the nextflow pipeline but couldn't figure out how to access it. 
This parses for bracken output filenames for 'pre_dehosting' or 'post_dehosting' in and appends respective strings to qc report fieldnames.

Pending: combining pre- and post-dehosting summaries side by side.

### CODE REVIEW

**Comment by dfornika** [comment]
Could you modify this:

https://github.com/BCCDC-PHL/dehost-and-verify-illumina/blob/efcc5f6438f18c1820ca83fc0d22fc5adac48e65/modules/parse_bracken_output.nf#L13-L16

...to include the `--analysis_stage` flag that you've added to `parse_bracken_output.py`?
like this:
```
parse_bracken_output.py \ 
       --host_name "${host_name}" \ 
       --pathogen_name "${pathogen_name}" \
        --analysis_stage ${analysis_stage} \
       ${bracken_output} > ${sample_id}_pathogen_host_read_summary_${analysis_stage}.tsv 
```

File: bin/parse_bracken_output.py [hunk 1]
```diff
@@ -49,6 +49,18 @@ def parse_bracken_output(path_to_bracken_output):
 
     return bracken_output
 
+def get_analysis_stage(path_to_bracken_output):
```
Comment (dfornika) [inline bin/parse_bracken_output.py:52]: I'd prefer to get this info from the `--analysis_stage` command-line flag.

**Review by dfornika — CHANGES_REQUESTED** [review]
I've made a suggestion in the main 'Conversation' for this PR on how to incorporate the new `--analysis_stage` flag into the workflow.

Let's do that, then use the value from the command-line flag.

**Comment by dfornika** [comment]
Supports #7

**Review by dfornika — APPROVED** [review]

**Comment by dfornika** [comment]
This is looking good @DiDigsDNA. If you'd like, we could merge this into master and you could continue the next steps in another pull-request. Or if you're already pushing ahead on this branch then you can continue pushing here. Feel free to request another review (or ping me on slack) when you'd like me to take another look.

**Comment by dfornika** [comment]
This is looking great @DiDigsDNA. The last step will be to collect all of the merged dehosting reports together into a single file. The Nextflow `collectFile` operator can be used for this:

https://www.nextflow.io/docs/latest/operator.html#collectfile

It has an option called `keepheader` that can be used to combine multiple files that share a common header.

**Comment by DiDigsDNA** [comment]
@dfornika 
There are now 4 datasets (the original plus 3 the following modified sets):
     fake1_R1.fastq.gz, fake1_R2.fastq.gz (top 10 reads of each removed)
     fake2_R1.fastq.gz, fake2_R2.fastq.gz (top 20 reads of each removed)
     fake3_R1.fastq.gz, fake3_R2.fastq.gz (bottom  30 reads of each removed)
The ran in the pipeline on sabin without error and all produced a <sampleID>_dehosting_report.tsv with the correct output format.

**Review by dfornika — APPROVED** [review]

**Pull request merged**